processor/memorylimiterprocessor/README.md

```diff
@@ -8,58 +8,60 @@
-
-When the memory usage is above the hard limit in addition to refusing the data the
-processor will forcedly perform garbage collection in order to try to free memory.
-
-When the memory usage drop below the soft limit, the normal operation is resumed (data
-will no longer be refused and no forced garbage collection will be performed).
-
-The difference between the soft limit and hard limits is defined via `spike_limit_mib`
-configuration option. The value of this option should be selected in a way that ensures
-that between the memory check intervals the memory usage cannot increase by more than this
-value (otherwise memory usage may exceed the hard limit - even if temporarily).
+the same data. The receivers may also apply backpressure to their own data sources
+in order to slow the inflow of data into the Collector, and to allow memory usage
+to go below the set limits.
+
+>Warning: Data will be permanently lost if the component preceding the memory limiter
+> in the telemetry pipeline does not correctly retry sending data after the
+> ConsumeLogs/Trace/Metrics function returns.
+> We consider such components to be incorrectly implemented.
+
+When the memory usage is above the hard limit the processor will additionally
+force garbage collection to be performed.
+
+Normal operation is resumed when memory usage drops below the soft limit, meaning data
+will no longer be refused and the processor won't force garbage collection to
+be performed.
+
+The difference between the soft limit and hard limit is defined via `spike_limit_mib`
+configuration option. The value of `spike_limit_mib` should be selected in a way
+that ensures that memory usage cannot increase by more than this value within a single
+memory check interval. Otherwise, memory usage may exceed the hard limit, even if temporarily.
 A good starting point for `spike_limit_mib` is 20% of the hard limit. Bigger
 `spike_limit_mib` values may be necessary for spiky traffic or for longer check intervals.
 
 Note that while the processor can help mitigate out of memory situations,
 it is not a replacement for properly sizing and configuring the
 collector. Keep in mind that if the soft limit is crossed, the collector will
 return errors to all receive operations until enough memory is freed. This may
-eventually result in dropped data since the receivers may not be able to hold back
-and retry the data indefinitely.
+eventually result in dropped data since the receivers may not be able to
+retry the data indefinitely.
 
-It is highly recommended to configure `ballastextension` as well as the
-`memory_limiter` processor on every collector. The ballast should be configured to
-be 1/3 to 1/2 of the memory allocated to the collector. The memory_limiter
+It is highly recommended to configure the `ballastextension` as well as the
+`memory_limiter` processor on every collector. The `ballastextension` should be configured to
```

Comment:
@mx-psi @dmitryax should we be recommending `GOMEMLIMIT` instead?

Reply:
Right. We should

Reply:
Indeed

Reply:
I've updated based on the recommendation in the deprecation notice of the ballast extension. Feedback is appreciated to ensure it's correct.